lib/src/browser_client.dart

```diff
@@ -52,16 +52,17 @@ class BrowserClient extends BaseClient {
     request.headers.forEach(xhr.setRequestHeader);
 
     var completer = Completer<StreamedResponse>();
+
+    // TODO(kevmoo): not sure what to do here...
+    // ignore: void_checks
     unawaited(xhr.onLoad.first.then((_) {
-      // TODO(nweiz): Set the response type to "arraybuffer" when issue 18542
-      // is fixed.
-      var blob = xhr.response as Blob ?? Blob([]);
+      var blob = xhr.response as Blob;
```

Comment:
According to the annotations on the `dart:html` types, `response` is never null.

Reply:
What do you mean by this? I think it probably is true that if `onLoad.first` doesn't have an error this would be non-null, but I don't know what you mean by "the annotations on the `dart:html` types".